strict golint check for CI

We now have a setup where certain packages have thresholds for non compliance with golint before an error is reported by our check. Remove those soft caps, everything should be linted. This means we'll have to fix some linting issues :)


Here's an example ci result that still reports a success:

```
[warning] 6 messages in 'github.com/mysteriumnetwork/node/identity': 
/go/src/github.com/mysteriumnetwork/node/identity/manager_fake.go:30:1: exported function NewIdentityManagerFake should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/identity/manager_interface.go:20:6: exported type Manager should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/identity/signer_fake.go:20:6: exported type SignerFake should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/identity/signer_fake.go:24:1: exported method SignerFake.Sign should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/identity/verifier_fake.go:20:6: exported type VerifierFake should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/identity/verifier_fake.go:22:1: exported method VerifierFake.Verify should have comment or be unexported
[warning] 11 messages in 'github.com/mysteriumnetwork/node/tequilapi': 
/go/src/github.com/mysteriumnetwork/node/tequilapi/api_test_utils.go:29:6: exported type TestClient should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/doc.go:18:1: package comment should be of the form "Package tequilapi ..."
/go/src/github.com/mysteriumnetwork/node/tequilapi/endpoints/sessions.go:28:1: comment on exported type SessionsDTO should be of the form "SessionsDTO ..." (with optional leading article)
/go/src/github.com/mysteriumnetwork/node/tequilapi/endpoints/sessions.go:33:1: comment on exported type SessionDTO should be of the form "SessionDTO ..." (with optional leading article)
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:35:1: exported method FieldErrorList.AddError should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:40:1: exported method FieldErrorList.MarshalJSON should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:44:6: exported type FieldErrorMap should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:48:1: exported function NewErrorMap should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:52:1: exported method FieldErrorMap.ForField should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:62:1: exported method FieldErrorMap.MarshalJSON should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/tequilapi/validation/validation.go:66:1: exported method FieldErrorMap.HasErrors should have comment or be unexported
[warning] 13 messages in 'github.com/mysteriumnetwork/node/service_discovery/dto': 
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/contact.go:33:6: exported type Contact should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/contact.go:38:6: exported type ContactDefinition should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/location.go:20:6: exported type Location should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/payment_method.go:22:6: exported type PaymentMethod should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_definition.go:20:6: exported type ServiceDefinition should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:31:6: exported type ServiceProposal should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:66:1: comment on exported type ServiceDefinitionUnserializer should be of the form "ServiceDefinitionUnserializer ..." (with optional leading article)
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:74:1: exported function RegisterServiceDefinitionUnserializer should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:85:1: comment on exported type PaymentMethodUnserializer should be of the form "PaymentMethodUnserializer ..." (with optional leading article)
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:93:1: exported function RegisterPaymentMethodUnserializer should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:105:1: comment on exported type ContactDefinitionUnserializer should be of the form "ContactDefinitionUnserializer ..." (with optional leading article)
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:113:1: exported function RegisterContactUnserializer should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/service_discovery/dto/service_proposal.go:152:1: exported method ServiceProposal.UnmarshalJSON should have comment or be unexported
[warning] 4 messages in 'github.com/mysteriumnetwork/node/money': 
/go/src/github.com/mysteriumnetwork/node/money/currency.go:20:6: exported type Currency should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/money/currency.go:23:2: exported const CURRENCY_MYST should have comment (or a comment on this block) or be unexported
/go/src/github.com/mysteriumnetwork/node/money/money.go:24:6: exported type Money should have comment or be unexported
/go/src/github.com/mysteriumnetwork/node/money/money.go:29:1: exported function NewMoney should have comment or be unexported

All packages are compliant to golint
```

--- a/bin/check_golint
+++ b/bin/check_golint
@@ -14,25 +14,6 @@
 
 source bin/helpers/output.sh
 
-check_uncleaned_package () {
-    PACKAGE=$1
-    PACKAGE_MESSAGES_MAX=$2
-
-    PACKAGE_MESSAGES=`echo "${MESSAGES_ERROR}" | grep "${PACKAGE}"`
-    PACKAGE_MESSAGES_COUNT=`echo -n "${PACKAGE_MESSAGES}" | grep -c '^'`
-    if [ ${PACKAGE_MESSAGES_COUNT} -gt 0 ]; then
-	print_warning "[warning] ${PACKAGE_MESSAGES_COUNT} messages in '${PACKAGE}': \n${PACKAGE_MESSAGES}\n"
-    fi
-
-    MESSAGES_ERROR=`echo -n "${MESSAGES_ERROR}" | grep -v "${PACKAGE}"`
-    if [ ${PACKAGE_MESSAGES_COUNT} -eq 0 ]; then
-        MESSAGES_RECONFIGURE+=("Congratulations! Cleaned all errors in package '${PACKAGE}'")
-    elif [ ${PACKAGE_MESSAGES_COUNT} -lt ${PACKAGE_MESSAGES_MAX} ]; then
-        MESSAGES_RECONFIGURE+=("Congratulations! Decreased ${PACKAGE_MESSAGES_MAX}->${PACKAGE_MESSAGES_COUNT} errors in package '${PACKAGE}'")
-    elif [ ${PACKAGE_MESSAGES_COUNT} -gt ${PACKAGE_MESSAGES_MAX} ]; then
-        MESSAGES_RECONFIGURE+=("Sad :(. Increased ${PACKAGE_MESSAGES_MAX}->${PACKAGE_MESSAGES_COUNT} errors in package '${PACKAGE}'")
-    fi
-}
 
 ARGUMENTS=$@
 if [ -z "$ARGUMENTS" ]; then
@@ -41,8 +22,6 @@ fi
 
 MESSAGES_ERROR=`golint --set_exit_status --min_confidence=1 ${ARGUMENTS} 2>/dev/null`
 MESSAGES_RECONFIGURE=()
-
-check_uncleaned_package "github.com/mysteriumnetwork/node/money" 4
 
 MESSAGES_ERROR_COUNT=`echo -n "${MESSAGES_ERROR}" | grep -c '^'`
 if [ ${MESSAGES_ERROR_COUNT} -gt 0 ]; then

--- a/cmd/di.go
+++ b/cmd/di.go
@@ -310,7 +310,7 @@ func newSessionManagerFactory(
 			// TODO: set the time and proper payment info
 			payment := dto.PaymentPerTime{
 				Price: money.Money{
-					Currency: money.CURRENCY_MYST,
+					Currency: money.CurrencyMyst,
 					Amount:   uint64(10),
 				},
 				Duration: time.Minute,

--- a/core/promise/methods/noop/promise_processor.go
+++ b/core/promise/methods/noop/promise_processor.go
@@ -71,7 +71,7 @@ type PromiseProcessor struct {
 func (processor *PromiseProcessor) Start(proposal market.ServiceProposal) error {
 	// TODO: replace static value with some real data
 	processor.lastPromise = promise.Promise{
-		Amount: money.NewMoney(10, money.CURRENCY_MYST),
+		Amount: money.NewMoney(10, money.CurrencyMyst),
 	}
 
 	consumer := promise.NewConsumer(proposal, processor.balance, processor.storage)

--- a/core/promise/methods/noop/promise_processor_test.go
+++ b/core/promise/methods/noop/promise_processor_test.go
@@ -53,7 +53,7 @@ func TestPromiseProcessor_Start_SendsBalanceMessages(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Exactly(
 		t,
-		promise.BalanceMessage{1, true, money.NewMoney(10, money.CURRENCY_MYST)},
+		promise.BalanceMessage{1, true, money.NewMoney(10, money.CurrencyMyst)},
 		lastMessage,
 	)
 }

--- a/money/currency.go
+++ b/money/currency.go
@@ -17,8 +17,10 @@
 
 package money
 
+// Currency represents a supported currency
 type Currency string
 
 const (
-	CURRENCY_MYST = Currency("MYST")
+	// CurrencyMyst is the myst token currency representation
+	CurrencyMyst = Currency("MYST")
 )

--- a/money/money.go
+++ b/money/money.go
@@ -21,11 +21,13 @@ import (
 	"fmt"
 )
 
+// Money holds the currency type and amount
 type Money struct {
 	Amount   uint64   `json:"amount,omitempty"`
 	Currency Currency `json:"currency,omitempty"`
 }
 
+// NewMoney returns a new instance of Money
 func NewMoney(amount float64, currency Currency) Money {
 	return Money{uint64(amount * 100000000), currency}
 }

--- a/money/money_test.go
+++ b/money/money_test.go
@@ -27,24 +27,24 @@ func Test_NewMoney(t *testing.T) {
 	assert.Equal(
 		t,
 		uint64(0.150*100000000),
-		NewMoney(0.150, CURRENCY_MYST).Amount,
+		NewMoney(0.150, CurrencyMyst).Amount,
 	)
 
 	assert.Equal(
 		t,
 		uint64(0*100000000),
-		NewMoney(0, CURRENCY_MYST).Amount,
+		NewMoney(0, CurrencyMyst).Amount,
 	)
 
 	assert.Equal(
 		t,
 		uint64(10*100000000),
-		NewMoney(10, CURRENCY_MYST).Amount,
+		NewMoney(10, CurrencyMyst).Amount,
 	)
 
 	assert.NotEqual(
 		t,
 		uint64(1),
-		NewMoney(1, CURRENCY_MYST).Amount,
+		NewMoney(1, CurrencyMyst).Amount,
 	)
 }

--- a/services/noop/discovery_dto_test.go
+++ b/services/noop/discovery_dto_test.go
@@ -26,7 +26,7 @@ import (
 )
 
 func Test_PaymentMethod_Serialize(t *testing.T) {
-	price := money.NewMoney(0.5, money.CURRENCY_MYST)
+	price := money.NewMoney(0.5, money.CurrencyMyst)
 
 	var tests = []struct {
 		model        PaymentNoop
@@ -60,7 +60,7 @@ func Test_PaymentMethod_Serialize(t *testing.T) {
 }
 
 func Test_PaymentMethod_Unserialize(t *testing.T) {
-	price := money.NewMoney(0.5, money.CURRENCY_MYST)
+	price := money.NewMoney(0.5, money.CurrencyMyst)
 
 	var tests = []struct {
 		json          string

--- a/services/noop/service.go
+++ b/services/noop/service.go
@@ -73,7 +73,7 @@ func GetProposal(country string) market.ServiceProposal {
 		},
 		PaymentMethodType: PaymentMethodNoop,
 		PaymentMethod: PaymentNoop{
-			Price: money.NewMoney(0, money.CURRENCY_MYST),
+			Price: money.NewMoney(0, money.CurrencyMyst),
 		},
 	}
 }

--- a/services/openvpn/discovery/dto/payment_method_per_bytes_test.go
+++ b/services/openvpn/discovery/dto/payment_method_per_bytes_test.go
@@ -27,7 +27,7 @@ import (
 )
 
 var (
-	price = money.NewMoney(0.5, money.CURRENCY_MYST)
+	price = money.NewMoney(0.5, money.CurrencyMyst)
 )
 
 func TestPaymentMethodPerBytesSerialize(t *testing.T) {

--- a/services/openvpn/discovery/dto/payment_method_per_time_test.go
+++ b/services/openvpn/discovery/dto/payment_method_per_time_test.go
@@ -27,7 +27,7 @@ import (
 )
 
 func TestPaymentMethodPerTimeSerialize(t *testing.T) {
-	price := money.NewMoney(0.5, money.CURRENCY_MYST)
+	price := money.NewMoney(0.5, money.CurrencyMyst)
 
 	var tests = []struct {
 		model        PaymentPerTime
@@ -64,7 +64,7 @@ func TestPaymentMethodPerTimeSerialize(t *testing.T) {
 }
 
 func TestPaymentMethodPerTimeUnserialize(t *testing.T) {
-	price := money.NewMoney(0.5, money.CURRENCY_MYST)
+	price := money.NewMoney(0.5, money.CurrencyMyst)
 
 	var tests = []struct {
 		json          string

--- a/services/openvpn/discovery/factory.go
+++ b/services/openvpn/discovery/factory.go
@@ -43,7 +43,7 @@ func NewServiceProposalWithLocation(
 		PaymentMethodType: dto.PaymentMethodPerTime,
 		PaymentMethod: dto.PaymentPerTime{
 			// 15 MYST/month = 0,5 MYST/day = 0,125 MYST/hour
-			Price:    money.NewMoney(0.125, money.CURRENCY_MYST),
+			Price:    money.NewMoney(0.125, money.CurrencyMyst),
 			Duration: 1 * time.Hour,
 		},
 	}

--- a/services/wireguard/service/service.go
+++ b/services/wireguard/service/service.go
@@ -132,7 +132,7 @@ func GetProposal(country string) market.ServiceProposal {
 		},
 		PaymentMethodType: wg.PaymentMethod,
 		PaymentMethod: wg.Payment{
-			Price: money.NewMoney(0, money.CURRENCY_MYST),
+			Price: money.NewMoney(0, money.CurrencyMyst),
 		},
 	}
 }

--- a/services/wireguard/wireguard_test.go
+++ b/services/wireguard/wireguard_test.go
@@ -26,7 +26,7 @@ import (
 )
 
 func Test_PaymentMethod_Serialize(t *testing.T) {
-	price := money.NewMoney(0.5, money.CURRENCY_MYST)
+	price := money.NewMoney(0.5, money.CurrencyMyst)
 
 	var tests = []struct {
 		model        Payment
@@ -60,7 +60,7 @@ func Test_PaymentMethod_Serialize(t *testing.T) {
 }
 
 func Test_PaymentMethod_Unserialize(t *testing.T) {
-	price := money.NewMoney(0.5, money.CURRENCY_MYST)
+	price := money.NewMoney(0.5, money.CurrencyMyst)
 
 	var tests = []struct {
 		json          string

--- a/session/amount_calculator_test.go
+++ b/session/amount_calculator_test.go
@@ -32,7 +32,7 @@ func Test_CorrectMoneyValueIsReturnedForTotalAmount(t *testing.T) {
 			Duration: time.Minute,
 			Price: money.Money{
 				Amount:   100,
-				Currency: money.CURRENCY_MYST,
+				Currency: money.CurrencyMyst,
 			},
 		},
 	}

--- a/session/balance/provider/balance_tracker_test.go
+++ b/session/balance/provider/balance_tracker_test.go
@@ -31,7 +31,7 @@ func Test_BalanceTracker(t *testing.T) {
 	mockTime := time.Second
 	mockMoney := money.Money{
 		Amount:   1,
-		Currency: money.CURRENCY_MYST,
+		Currency: money.CurrencyMyst,
 	}
 	mtk := &mockTimeKeeper{elapsed: mockTime}
 	mac := &mockAmountCalculator{toReturn: mockMoney}

